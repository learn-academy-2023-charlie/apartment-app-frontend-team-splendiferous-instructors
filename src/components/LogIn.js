// import
import React from "react"
// declare functional component
const LogIn = () => {
  return(
    <>
      <h3>LogIn component</h3>
    </>
  )
}
// export
export default LogIn