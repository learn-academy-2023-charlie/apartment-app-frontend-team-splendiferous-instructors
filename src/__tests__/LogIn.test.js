import { render, screen } from '@testing-library/react';
import LogIn from '../components/LogIn';
import { BrowserRouter } from 'react-router-dom'

describe("<LogIn />", () => {
  beforeEach(() => {
    render(
      <BrowserRouter>
        <LogIn />
      </BrowserRouter>
    )
  })
  // ensures react component renders
  it('renders without crashing', () => {})

  // testing an element of the UI
  it('renders an email field on the form', () => {
    // debugging tool
    // screen.logTestingPlaygroundURL()
    // query
    const logIn = screen.getByText(/email:/i)
    // screen.debug(logIn) 
    // assertion
    expect(logIn).toBeInTheDocument()
    expect(logIn).toHaveTextContent(/email:/i)
  })

  it('renders a Signup link on the form', () => {
    // query
    const signUp = screen.getByText(/not registered yet,/i) 
    // assertion
    expect(signUp).toBeInTheDocument()
    expect(signUp).toHaveTextContent(/signup/i)

  })

  it('renders a password field on the form', () => {
    // query
    const passWord = screen.getByPlaceholderText(/password/i) 
    // assertion
    expect(passWord).toBeInTheDocument()
    expect(passWord).toMatchSnapshot()
  })
})