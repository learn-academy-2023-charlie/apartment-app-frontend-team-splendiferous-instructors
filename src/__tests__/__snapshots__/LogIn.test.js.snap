// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn /> renders a password field on the form 1`] = `
<input
  name="password"
  placeholder="password"
  type="password"
/>
`;
